acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/pageobjects/BitbucketBranchSource.java

```diff
@@ -29,6 +34,26 @@ public BitbucketBranchSource credentialsId(String credentialsId) {
         return this;
     }
 
+    public BitbucketBranchSource discoverBranches(boolean isDiscoveryEnabled) {
+        if (isDiscoveryEnabled) {
+            addTrait("Discover branches");
```

Comment:
Is this idempotent?  If it is already present it does not fail? I assume it does not?

Reply:
I don't think it will fail. Clicking on an option that's already selected just won't do anything. In any case, we have an isEnabled check which prevents clicking on it if already selected.